### DISCUSSION TITLE
How does \cite fetch its items?

### DISCUSSION THREAD

**Discussion by Atcold** [texstudio-org/texstudio#3766] [Q&A] [answered]
I understand that `\cite` looks at the presence of `\bibliography{}` to autocomplete its argument and highlight in green correct options.
I've been using a macro which contains the two commands `\bibliographystyle{alpha}\bibliography{main}`, since I'm adding them to each chapter of my book. Doing this, `\cite` will (obviously) fail to perform any check. Is there a workaround for informing `\cite` where to source its info?

Also, how to open `\cite`'s autocompletion list *before* typing anything?

**Comment by sunderme** [comment, marked as answer]
Txs looks for the files in the command bibliography. I am not aware of alternatives.

**Reply by Atcold** [reply]
But such file needs to be already processed by BibTeX, correct?
I'm adding sources, and they don't appear right away on the autocompletion list.
Also, is there a way to open the autocompletion without typing a character first?

**Reply by muzimuzhi** [reply]
> Also, is there a way to open the autocompletion without typing a character first?

Menu "Idefix - Complete - Normal" with default shortcut `Meta + Space` which can be configured in "Configuration - Shortcuts".

**Reply by sunderme** [reply]
txs reads bib files, no preprocessing required.

**Comment by Atcold** [comment]
I'm trying to issue the "Idefix - Complete - Normal" command after tpying `\cite`⏎.
This gets completed with `\cite{bibid}`.
Now, pressing `^`␣ does nothing, because it tries to complete `bibid`.
Is there a workaround? Do I have to manually erase `bibid`?

**Comment by sunderme** [comment]
I am at a loss what you are doing.
With default configuration, `\cite{bibid}`is the result of completion and `bibid`is selected. Hence `bibid`is replaced with whatever you enter and the completer opens automatically to suggest matching ids.
If you want to start to see all bibids, pressing "del" removes the selection and you can open the completer with ctrl+space (or meta+space)
Furthermore while completing `\cite`, variants with `\cite{specific ids}` are offered.

**Reply by Atcold** [reply]
Right right.

> If you want to start to see all bibids, pressing "del" removes the selection and you can open the completer with ctrl+space (or meta+space)

Three keys need to be pressed. Just double checking. 

> Furthermore while completing `\cite`, variants with `\cite{specific ids}` are offered.

I didn’t understand this statement.
